src/BPEssentials/FileChecker.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace BPEssentials
+{
+    public static class FileChecker
+    {
+        public static readonly HttpClient client = new HttpClient();
+
+        public static async Task CheckFiles()
+        {
+
+            if (!Directory.Exists(Paths.EssentialsFolder))
+            {
+                Core.Instance.Logger.LogWarning("Essentials Folder not found.");
+                Directory.CreateDirectory(Paths.EssentialsFolder);
+                Core.Instance.Logger.LogInfo("Created Essentials Folder.");
+            }
+            foreach (var file in RequiredFilesDictionary)
+            {
+                if (File.Exists(file.Value)) continue;
```

Comment:
Please put `{` and `}` on new lines. It doesn't look as good, just makes it clearer to read.

```suggestion
                if (File.Exists(file.Value))
                {
                      continue;
                }
```